draft-ietf-tls-esni.md

```diff
@@ -277,7 +277,16 @@ of a particular ECH configuration.
 public_name
 : The non-empty name of the client-facing server, i.e., the entity trusted to
 update the ECH configuration. This is used to correct misconfigured clients, as
-described in {{handle-server-response}}.
+described in {{handle-server-response}}. Clients MUST ignore any `ECHConfig`
+structure with a `public_name` that begins or ends with an ASCII dot or cannot
+be parsed as a dot-separated sequence of LDH labels, as defined in {{RFC5890}},
+Section 2.3.1. This validation procedure is incomplete; it incidentally rules
+out textual representations of IPv6 addresses (see {{RFC3986}}, Section 3.2.2),
+but it does not exclude IPv4 addresses in standard dotted-decimal or other
+non-standard notations such as octal and hexadecimal (see {{RFC3986}}, Section
+7.4). If `public_name` contains a literal IPv4 or IPv6 address, the client
+SHOULD ignore the `ECHConfig` to avoid sending a non-compliant "server_name"
+extension on the ClientHelloOuter (see {{RFC6066, Section 3}}).
```

Comment:
Man, this IP address thing is a mess. The SHOULD makes me sad, but I don't have a clear alternate suggestion. I can see not wanting to tie to the slightly odd "first-match-wins" parse from URLs. (And then there's the mess that is URL parsing...) Although, amusingly, RFC6066 section 3 clearly assumes that we have somehow defined this in a TLS-compatible way, but it forgot to cite anything. :-(

I dunno, I'll defer to others on the exact wording here. @martinthomson, what do you think?

Either way, I'm on board with dealing with one problem at a time in this PR.